actionpack/test/dispatch/request_test.rb

```diff
@@ -1290,3 +1290,21 @@ def setup
     assert_equal expected_hints, early_hints
   end
 end
+
+class RequestInspectTest < BaseRequestTest
+  test "inspect" do
+    request = stub_request(
+      "REQUEST_METHOD" => "POST",
+      "REMOTE_ADDR" => "1.2.3.4",
+      "HTTP_X_FORWARDED_PROTO" => "https",
+      "HTTP_X_FORWARDED_HOST" => "glu.ttono.us:443",
+      "PATH_INFO" => "/path/of/some/uri",
+      "QUERY_STRING" => "mapped=1",
+      "CONTENT_TYPE" => "application/x-www-form-urlencoded; charset=utf-8",
+    )
+    assert_match %r(#<ActionDispatch::Request method="POST"), request.inspect
+    assert_match %r(original_url="https://glu.ttono.us/path/of/some/uri\?mapped=1"), request.inspect
+    assert_match %r(remote_ip="1.2.3.4"), request.inspect
+    assert_match %r(media_type="application/x-www-form-urlencoded">), request.inspect
```

Comment:
```suggestion
    assert_equal '#<ActionDispatch::Request method="POST", original_url="https://glu.ttono.us/path/of/some/uri?mapped=1", remote_ip="1.2.3.4", media_type="application/x-www-form-urlencoded">', request.inspect
```

Small suggestion but I think we can just test the entire string without having to regexp match on parts of it.

Reply:
I think it's kind of long. Maybe the output of inspect can be shortened to just:
`#<ActionDispatch::Request POST https://glu.ttono.us/path/of/some/uri?mapped=1>`

Reply:
@tgxworld I've shortend the output to same format as used for requests in the logs. This also allows the assertion to use a single line without getting to long.

        "#<ActionDispatch::Request POST "https://example.com/path/of/some/uri?q=1" for 1.2.3.4>"